### PR TITLE
feat: add CIS Docker Benchmark scanner

### LINKED ISSUE #1254: Add CIS Docker Benchmark misconfiguration scanner
## Problem Statement

Migrate CIS Docker Benchmark from KubeClarity to VMClarity under the [misconfiguration scanner family](https://github.com/openclarity/vmclarity/blob/84c711c0246c4d8aab7c2cb6f55fd35a7c741118/cli/families/misconfiguration). The new scanner should be able to scan containers and container images. 

## Proposed Solution

Use [Lynis scanner](https://github.com/openclarity/vmclarity/blob/84c711c0246c4d8aab7c2cb6f55fd35a7c741118/cli/families/misconfiguration/lynis) as an implementation reference.

### PR DESCRIPTION
**Pull request by akijakya**
## Description

This PR adds `dockle` as CIS Docker Benchmark scanner to the family of Misconfiguration scanners.

### Addressing [fix: go builds](https://github.com/openclarity/vmclarity/pull/1303/commits/99677854e48bc06b7a64d99301924d28162d6f8e) commit

Additional build tags had to be added everywhere where the `cli` and modules using the `cli` module (namely `orchestrator` and `provider`) are being built (because of the `github.com/containers/storage` indirect package) and CGO is enabled, otherwise, this build error would occur:

```shell
0.926 # github.com/containers/storage/pkg/devicemapper
0.926 # [aarch64-linux-gnu-pkg-config --cflags  -- devmapper]
0.926 Please install dpkg-dev to use pkg-config when cross-building
0.950 # github.com/containers/storage/drivers/btrfs
0.950 /go/pkg/mod/github.com/containers/storage@v1.51.0/drivers/btrfs/btrfs.go:12:10: fatal error: btrfs/ioctl.h: No such file or directory
0.950    12 | #include <btrfs/ioctl.h>
0.950       |          ^~~~~~~~~~~~~~~
0.950 compilation terminated.
```

If this is unacceptable, we could think of other ways to include this scanner, e.g., adding it as a binary instead of using it as a go package.

### Try it out locally

1. Get a test image (e.g. `docker pull goodwithtech/test-image:v1`, or any image on your system you'd like to scan)
2. Also, since Dockle can scan docker archives as well, let's create one, e.g.: `docker buildx build --output type=docker,dest=./docker-test.tar --build-arg BUILD_OPTS="-tags=exclude_graphdriver_btrfs,exclude_graphdriver_devicemapper" -f Dockerfile.cli .`
3. Create a config with these inputs:

    ```shell
    cat <<EOF >> .cisdocker-test.yaml
    misconfiguration:
      enabled: true
      scanners_list:
        - "cisdocker"
      inputs:
        - input: "goodwithtech/test-image:v1"
          input_type: "image"
        - input: "docker-test.tar"
          input_type: "docker-archive"
    EOF
    ```
4. Run the scan: `go run ./cli/cmd/main.go scan --config .cisdocker-test.yaml`

For trying it out locally as part of the docker provider within VMClarity, [this](https://github.com/openclarity/vmclarity/issues/1304) issue needs to be resolved first.

Fixes #1254 

## Type of Change

[ ] Bug Fix  
[X] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [X] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [X] Existing issues have been referenced (where applicable)
- [X] I have verified this change is not present in other open pull requests
- [X] Functionality is documented
- [X] All code style checks pass
- [X] New code contribution is covered by automated tests
- [X] All new and existing tests pass

### CODE REVIEW

File: cli/families/misconfiguration/cisdocker/scanner.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			return
+		}
+
+		assessmentMap, err := dockle_run.RunFromConfig(createDockleConfig(a.logger, a.config, userInput))
+		if err != nil {
+			a.sendResults(retResults, fmt.Errorf("failed to run dockle: %w", err))
+			return
+		}
+
+		a.logger.Infof("Image was scanned.")
```
Comment (ramizpolic) [inline cli/families/misconfiguration/cisdocker/scanner.go:71]:
```suggestion
		a.logger.Infof("Successfully scanned %s %s", sourceType, userInput)
```

File: .golangci.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 run:
   timeout: 20m
+  build-tags:
+    - exclude_graphdriver_btrfs
+    - exclude_graphdriver_devicemapper
```
[the remark this answers is not in the document]
Comment (akijakya) [reply .golangci.yml:5]: great idea, thanks!

[the remark this answers is not in the document]
Comment (akijakya) [reply .golangci.yml:5]: great idea, thanks!

File: .github/workflows/build-and-push-component.yaml [hunk 3]
```diff
@@ -96,6 +96,7 @@ jobs:
           VERSION: ${{ inputs.image_tag }}
           BUILD_TIMESTAMP: ${{ inputs.timestamp }}
           COMMIT_HASH: ${{ github.sha }}
+          BUILD_OPTS: -tags=exclude_graphdriver_btrfs,exclude_graphdriver_devicemapper
```
[the remark this answers is not in the document]
Comment (ramizpolic) [reply .github/workflows/build-and-push-component.yaml:99]: created https://github.com/openclarity/vmclarity/issues/1368

File: cli/families/misconfiguration/cisdocker/report.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		if severity != "" {
+			ret = append(ret, types.Misconfiguration{
+				Location:    imageName,
+				Category:    "best-practice",
+				ID:          codeInfo.Code,
+				Description: mergeDockleDescriptions(codeInfo.Assessments),
+				Severity:    severity,
+				Message:     dockle_types.TitleMap[codeInfo.Code],
+			})
+		}
```
Comment (ramizpolic) [inline cli/families/misconfiguration/cisdocker/report.go:38]:
```suggestion
		severity := convertDockleSeverity(codeInfo.Level)
		if severity == "" {
			// skip when no severity
			continue
		}
		
		description := ""
		for _, assessment := range codeInfo.Assessments {
			description += assessment.Desc + "\n"
		}
		
		ret = append(ret, types.Misconfiguration{
			Location:    imageName,
			Category:    "best-practice",
			ID:          codeInfo.Code,
			Description: description,
			Severity:    severity,
			Message:     dockle_types.TitleMap[codeInfo.Code],
		})
```

File: cli/families/misconfiguration/cisdocker/report.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	}
+
+	return ret
+}
+
+func mergeDockleDescriptions(assessments dockle_types.AssessmentSlice) string {
```
Comment (ramizpolic) [inline cli/families/misconfiguration/cisdocker/report.go:44]: added above, we can remove this one too

File: cli/families/misconfiguration/cisdocker/report.go (hunk 4 above)
Comment (akijakya) [reply cli/families/misconfiguration/cisdocker/report.go:38]: Looks cleaner, thanks for the suggestion!

**Review by ramizpolic — APPROVED** [review]
lgtm, thanks @akijakya!

**Pull request merged**